plugins/kubernetes/app/models/kubernetes/blue_green_deploy_executor.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Kubernetes
+  class BlueGreenDeployExecutor < DeployExecutor
+    private
+
+    # overrides
+    def deploy_and_watch(release, release_docs)
+      raise "does not support prerequisites" if release.release_docs != release_docs
+
+      deploy_resources(release_docs)
+
+      result = wait_for_resources_to_complete(release, release_docs)
+      if result == true
+        switch_service(release_docs)
+        previous = release.previous_successful_release
+        delete_resources(previous.release_docs) if previous && previous.blue_green_color != release.blue_green_color
+        success
+      else
+        show_failure_cause(release, release_docs, result)
+        delete_resources(release_docs)
+        @output.puts "DONE"
+        false
+      end
+    end
+
+    def deploy_resources(release_docs)
+      release_docs.each do |release_doc|
+        @output.puts "Creating #{release_doc.blue_green_color.upcase} " \
+          "resources for #{release_doc.deploy_group.name} role #{release_doc.kubernetes_role.name}"
+        other_resources(release_doc).each(&:deploy)
+      end
+    end
+
+    def switch_service(release_docs)
+      release_docs.each do |release_doc|
+        next unless service = service(release_doc)
+        @output.puts "Switching service for #{release_doc.deploy_group.name} " \
+          "role #{release_doc.kubernetes_role.name} to #{release_doc.blue_green_color.upcase}"
+        service.deploy
```

Comment:
What happens if this fails?

Reply:
keeps pointing at the old resources and deploy fails

Reply:
Maybe rescue exceptions and call `delete_resources(release_docs)`?